example-wasm/README.md

```diff
@@ -0,0 +1,200 @@
+# Example React App Using Realm & Sync for Web
+
+This is an example React Todo/Task app for showcasing Realm and Sync for Web.
+
+## MongoDB & Realm Functionality
+
+### Use cases
+
+* Log in and register (email/password)
+* Log out
+* Create tasks
+* Read/query tasks
+* Update the status of tasks
+* Delete tasks
+* Sync
+  1. Tasks are stored locally in an in-memory realm..
+  2. then synced to MongoDB Atlas..
+  3. then synced to all other apps connected to the same Atlas App.
+* Local/Offline-first
+  * All CRUD functionality works while offline.
+* Realm JS and [@realm/react](https://www.npmjs.com/package/@realm/react) hooks
+
+### Screenshot
+
+![Tasks Page](./src/assets/screenshot-realm-web-sync-tasks.png)
+
+## Project Structure
+
+The following shows the project structure and the most relevant files.
+
+```
+├── public
+│   └── index.html              - File served to client
+│
+├── src
+│   ├── atlas-app-services
+│   │   └── config.json         - Set Atlas App ID
+│   │
+│   ├── components
+│   │   ├── AddTaskForm.tsx     - Trigger create task
+│   │   ├── NavBar.tsx          - Trigger logout
+│   │   ├── TaskItem.tsx        - Trigger update/delete task
+│   │   └── TaskList.tsx        - Render all tasks
+│   │
+│   ├── hooks
+│   │   ├── useAppManager.ts    - Handle login/register
+│   │   └── useTaskManager.ts   - Handle CRUD task
+│   │
+│   ├── models
+│   │   └── Task.ts             - Data model
+│   │
+│   ├── pages
+│   │   ├── LoginPage.tsx       - Trigger login/register
+│   │   └── TaskPage.tsx        - Pass CRUD ops to children
+│   │
+│   ├── index.tsx               - Entry point
+│   ├── App.tsx                 - Get and provide Atlas App
+│   └── AuthenticatedApp.tsx    - Open and provide Realm & User
+│
+├── craco.config.ts             - Configure CRA
+├── package.json                - Specify Node dependencies
+└── tsconfig.json               - Configure TypeScript
+```
+
+## Getting Started
+
+### Prerequisites
+
+* Emscripten v3.1.40 or later.
+  * Follow the [recommended installation instructions](https://emscripten.org/docs/getting_started/downloads.html#installation-instructions-using-the-emsdk-recommended).
```

Comment:
It took a bit of digging in these instructions, but `brew install emscripten` works just fine for mac users.  Might want to just provide that step here to save some time.

Reply:
Main reason for this was that the `brew install emscripten` did not necessarily install the latest version and is an unofficial package, as opposed to the emsdk (the current instruction) which is the officially supported package.

Using homebrew, emscripten v3.1.39 was installed and I encountered a time consuming (but interesting 😌) bug that @nhachicha helped debug. But like discussed on Slack, perhaps a `brew update && brew install emscripten` could be a helpful replacement of the current instructions? Also, once we can provide an npm realm wasm package, we can hopefully abstract away the emscripten part.

Reply:
At some point we will have an actual template with this code instead of the example.  In that case, these instructions would be for the person (someone debugging or testing WASM features) trying to use the example.  Installing `emscripten` is a bit like us having to install `cmake`, but our consumers don't have to worry about it.

Reply:
Agreed, with the now released realm package w/ the browser target I'll make some updates, along with an update to the @realm/react dependency once a release compatible with v12 is done 👍